### PR TITLE
fix name deduced from URL

### PR DESCRIPTION
**Pull request by ClausKlein**

### CODE REVIEW

**Comment by ClausKlein** [comment]
@TheLartians would you like to review this MR?

File: cmake/CPM.cmake [hunk 1]
```diff
@@ -167,14 +167,30 @@ include(FetchContent)
...
 # Try to infer package name from git repository uri (path or url)
 function(cpm_package_name_from_git_uri URI RESULT)
-  if("${URI}" MATCHES "([^/:]+)/?.git/?$")
-    set(${RESULT}
-        ${CMAKE_MATCH_1}
-        PARENT_SCOPE
-    )
-  else()
-    unset(${RESULT} PARENT_SCOPE)
+  message(DEBUG "${URI}")
```
Comment (TheLartians) [inline cmake/CPM.cmake:170]: Please remove any debug log messages in your code.

File: test/unit/package_name_from_git_uri.cmake [hunk 2]
```diff
@@ -6,15 +6,24 @@ include(${CPM_PATH}/testing.cmake)
...
 
 cpm_package_name_from_git_uri("git://host.xz/path/to/pkg.git" name)
 assert_equal("pkg" ${name})
 
-cpm_package_name_from_git_uri("git@host.xz:cool-pkg.git" name)
+cpm_package_name_from_git_uri("git@host.xz:./cool-pkg.git" name)
 assert_equal("cool-pkg" ${name})
 
+cpm_package_name_from_git_uri("git@github.com:user/repo.git" name)
+assert_equal("repo" ${name})
```
Comment (TheLartians) [inline test/unit/package_name_from_git_uri.cmake:25]: All test cases you added seem to work fine using the old code as well.

**Review by TheLartians — COMMENTED** [review]
I'm not sure I understand the changes of this PR, as the existing regex seems to be working fine for me for the mentioned test cases here and in #407.

File: cmake/CPM.cmake [hunk 3]
```diff
@@ -167,14 +167,30 @@ include(FetchContent)
...
+      # ssh://user@host.xz:123/path/to/pkg.git/
+      set(${RESULT}
+          ${CMAKE_MATCH_2}
+          PARENT_SCOPE
+      )
+      message(DEBUG "${CMAKE_MATCH_2}")
+      return()
+    else()
+      # i.e. ../local-repo/.git
+      if("${URI}" MATCHES "([^/:]+)/?.git/?$")
```
Comment (ClausKlein) [inline cmake/CPM.cmake:183]: **I dit not and do not understand this regex?**

I was searching for a BUG that does not exist.
sorry

File: test/unit/package_name_from_git_uri.cmake [hunk 4]
```diff
@@ -6,15 +6,24 @@ include(${CPM_PATH}/testing.cmake)
...
+cpm_package_name_from_git_uri("https://github.com/aminya/project_options.git" name)
+assert_equal("project_options" ${name})
+
 cpm_package_name_from_git_uri("ssh://user@host.xz:123/path/to/pkg.git/" name)
 assert_equal("pkg" ${name})
 
 cpm_package_name_from_git_uri("git://host.xz/path/to/pkg.git" name)
 assert_equal("pkg" ${name})
 
-cpm_package_name_from_git_uri("git@host.xz:cool-pkg.git" name)
```
Comment (ClausKlein) [inline test/unit/package_name_from_git_uri.cmake:15]: is this really a realistic URL?

File: test/unit/package_name_from_git_uri.cmake (hunk 2 above)
Comment (ClausKlein) [reply test/unit/package_name_from_git_uri.cmake:25]: yes

**Comment by ClausKlein** [comment]
Sorry to waste your time.

**Pull request closed without merging**